hurricane/server/django.py

```diff
@@ -19,52 +20,101 @@ def prepare(self) -> None:
+            except SystemCheckError as e:
+                if settings.DEBUG:
+                    self.write("django check error: " + str(e))
+                else:
+                    self.write("check error")
+                self.set_status(500)
+            except OperationalError as e:
+                if settings.DEBUG:
+                    self.write("django database error: " + str(e))
+                else:
+                    self.write("db error")
+                self.set_status(500)
             else:
-                self.write("check error")
-            self.set_status(500)
-        except OperationalError as e:
-            if settings.DEBUG:
-                self.write("django database error: " + str(e))
-            else:
-                self.write("db error")
-            self.set_status(500)
+                if response_average_time := ResponseTimeAverageMetric.get():
+                    self.write(
+                        f"Average response time: {response_average_time:.2f}ms Request "
+                        f"queue size: {RequestQueueLengthMetric.get()} Rx"
+                    )
+                else:
+                    self.write("alive")
         else:
-            if response_average_time := ResponseTimeAverageMetric.get():
-                self.write(
-                    f"Average response time: {response_average_time:.2f}ms Request "
-                    f"queue size: {RequestQueueLengthMetric.get()} Rx"
-                )
+            self.set_status(400)
+
+
+class DjangoReadinessHandler(DjangoProbeHandler):
+    """
+    This handler runs with every call to the probe endpoint which is supposed to be used
+    with Kubernetes 'Readiness Probes'. The DjangoCheckHandler calls Django's Check Framework which
+    can be used to determine the application's health state during its operation.
+    """
+
+    def initialize(self, req_queue_len):
+        self.request_queue_length = req_queue_len
+
+    def _check(self):
+        if StartupTimeMetric.get():
+            if RequestQueueLengthMetric.get() > self.request_queue_length:
+                self.set_status(400)
             else:
-                self.write("alive")
+                self.set_status(200)
+        else:
+            self.set_status(400)
 
-    def set_extra_headers(self, path):
-        self.set_header("Cache-Control", "no-store, no-cache, must-revalidate, max-age=0")
 
-    def get(self):
-        self._check()
+class DjangoStartupHandler(DjangoProbeHandler):
+    """
+    This handler runs with every call to the probe endpoint which is supposed to be used with Kubernetes
+    'Startup Probes'. It returns 400 response for post and get requests, if StartupTimeMetric is not set, what means
+    that the application is still in the startup phase. As soon as StartupTimeMetric is set, this handler returns 200
+    response upon request, it indicated, that startup phase is finished and Kubernetes can now poll
```

Comment:
Following sounds more natural to me.
```suggestion
    response upon request, which indicates, that startup phase is finished and Kubernetes can now poll
```